### PR TITLE
build: make the package to support Python 3.11

### PR DESCRIPTION
**Pull request by marcofavoritobi**
## Proposed changes

This PR adds Python 3.11 in the list of supported Python versions. It also sets up the CI to also run tests over Python 3.11.

This PR has highest priority over PR #50; once this one is merged, that PR will be updated accordingly.

This PR should be merged after #72.

### CODE REVIEW

**Review by muxator — APPROVED** [review]
LGTM once we merge the modified #72.

**Review by AldoGl — APPROVED** [review]

**Pull request merged**
